File: Coding_Problems_Java/Maximum_score_after_splitting_a_string.java
class Solution {
    public int maxScore(String s) {

        int maxi = 0, left = 0;
        int right = Count(s, '1');
        
        for (int i = 0; i < s.length()-1; i++){
            if (s.charAt(i) == '0') left ++;
            else right --;
            maxi = Math.max(maxi, left + right);
        }
        return maxi;
    }
        
    public int Count(String  s, char target){
            int count = 0;
            for (char c: s.toCharArray()){
                if (c == target){
                    count ++;
                }
            }
            return count;
        }
        
    }

File: Coding Problems - Python/String/Maximum_score_after_splitting_a_string.py
class Solution:
    def maxScore(self, s: str) -> int:
        left, maxi, right = 0, 0, s.count("1")

        for i in range(len(s)-1):
            left += s[i] == "0"
            right -= s[i] == "1"
            maxi = max(maxi, left + right)

        return maxi
